examples/model_selection/plot_roc.py

```diff
@@ -1,132 +1,269 @@
+# consists in computing one ROC curve per class to evaluate the classification
+# scores of one class compared to all the other classification scores predicted
+# by the multiclass classifier.
+#
+# .. note:: One should not confuse the OvR strategy used to apply ROC
+#     **evaluation** to multiclass classifiers with the OvR strategy used to
+#     **train** a multiclass classifier by fitting a set binary classifiers (for
+#     instance via the :class:`~sklearn.multiclass.OneVsRestClassifier`
+#     meta-estimator). The OvR ROC evaluation can be used to evaluate any kind
+#     of classification models irrespectively of how they were trained (see
+#     :ref:`multiclass`).
+#
+# In this section we use a :class:`~sklearn.preprocessing.LabelBinarizer` to
+# binarize the target by one-hot-encoding in a OvR fashion. This means that the
+# target of shape (`n_samples`,) is mapped to a target of shape (`n_samples`,
+# `n_classes`).
+
+from sklearn.preprocessing import LabelBinarizer
+
+label_binarizer = LabelBinarizer().fit(y_train)
+y_onehot_test = label_binarizer.transform(y_test)
+y_onehot_test.shape  # (n_samples, n_classes)
+
+# %%
+# We can as well easily check the encoding of a specific class:
 
+label_binarizer.transform(["virginica"])
 
 # %%
-# Plot of a ROC curve for a specific class
-plt.figure()
-lw = 2
-plt.plot(
-    fpr[2],
-    tpr[2],
+# ROC curve showing a specific class
+# ----------------------------------
+#
+# In the following plot we show the resulting ROC curve when regarding the iris
+# flowers as either "virginica" (`class_id=2`) or "non-virginica" (the rest).
+
+class_of_interest = "virginica"
+class_id = np.flatnonzero(label_binarizer.classes_ == class_of_interest)[0]
+class_id
+
+# %%
+import matplotlib.pyplot as plt
+from sklearn.metrics import RocCurveDisplay
+
+RocCurveDisplay.from_predictions(
+    y_onehot_test[:, class_id],
+    y_score[:, class_id],
+    name=f"ROC curve for {class_of_interest} vs the rest",
     color="darkorange",
-    lw=lw,
-    label="ROC curve (area = %0.2f)" % roc_auc[2],
 )
-plt.plot([0, 1], [0, 1], color="navy", lw=lw, linestyle="--")
+plt.plot([0, 1], [0, 1], "k--", label="ROC curve for chance level (AUC = 0.5)")
 plt.xlim([0.0, 1.0])
-plt.ylim([0.0, 1.05])
+plt.ylim([0.0, 1.0])
+plt.axis("square")
 plt.xlabel("False Positive Rate")
 plt.ylabel("True Positive Rate")
-plt.title("Receiver operating characteristic example")
-plt.legend(loc="lower right")
+plt.title("One-vs-Rest ROC: Virginica vs (Setosa & Versicolor)")
+plt.legend()
+plt.show()
+
+# %%
+# ROC curve using micro-averaged OvR
+# ----------------------------------
+#
+# Micro-averaging aggregates the contributions from all the classes (using
+# :func:`np.ravel`) to compute the average metrics as follows:
+#
+# :math:`TPR=\frac{\sum_{c}TP_c}{\sum_{c}(TP_c + FN_c)}`
+#
+# :math:`FPR=\frac{\sum_{c}FP_c}{\sum_{c}(FP_c + TN_c)}`
+
+print(f"y_score: {y_score[0:2,:]}")
+print(f"y_score.ravel(): {y_score[0:2,:].ravel()}")
```

Comment:
Let's avoid printing several things in a given cell to make it straightforward to match code snippets to outputs:

```suggestion
# %%
print(f"y_score.ravel(): {y_score[0:2,:].ravel()}")
```